src/dbus.c

```diff
@@ -240,28 +227,28 @@ static notification *dbus_message_to_notification(const gchar *sender, GVariant
                                          * So let's check for int and boolean until notify-send is fixed.
                                          */
                                         if((dict_value = g_variant_lookup_value(content, "transient", G_VARIANT_TYPE_BOOLEAN))) {
-                                                transient = g_variant_get_boolean(dict_value);
+                                                n->transient = g_variant_get_boolean(dict_value);
                                                 g_variant_unref(dict_value);
                                         } else if((dict_value = g_variant_lookup_value(content, "transient", G_VARIANT_TYPE_UINT32))) {
-                                                transient = g_variant_get_uint32(dict_value) > 0;
+                                                n->transient = g_variant_get_uint32(dict_value) > 0;
                                                 g_variant_unref(dict_value);
                                         } else if((dict_value = g_variant_lookup_value(content, "transient", G_VARIANT_TYPE_INT32))) {
-                                                transient = g_variant_get_int32(dict_value) > 0;
+                                                n->transient = g_variant_get_int32(dict_value) > 0;
                                                 g_variant_unref(dict_value);
                                         }
 
                                         if((dict_value = g_variant_lookup_value(content, "value", G_VARIANT_TYPE_INT32))) {
-                                                progress = g_variant_get_int32(dict_value);
+                                                n->progress = g_variant_get_int32(dict_value);
                                                 g_variant_unref(dict_value);
                                         } else if((dict_value = g_variant_lookup_value(content, "value", G_VARIANT_TYPE_UINT32))) {
-                                                progress = g_variant_get_uint32(dict_value);
+                                                n->progress = g_variant_get_uint32(dict_value);
                                                 g_variant_unref(dict_value);
                                         }
                                 }
                                 break;
                         case 7:
                                 if (g_variant_is_of_type(content, G_VARIANT_TYPE_INT32))
-                                        timeout = g_variant_get_int32(content);
+                                        n->timeout = g_variant_get_int32(content) * 1000;
```

Comment:
This is also removing the sanity check that if `timeout < 0` then `timeout = -1`, so timeouts less than -1 are no undefined behaviour.

Reply:
`notification_init` should handle this:

https://github.com/dunst-project/dunst/blob/f1dd772937686d704dc943d4b0e5b074fd76792a/src/notification.c#L327-L329